sway-lib-std/src/storage.sw

```diff
@@ -236,6 +236,31 @@ impl<V> StorageVec<V> {
         element_to_be_removed
     }
 
+    /// Sets/mutates the value at the given index
+    ///
+    /// # Arguments
+    /// 
+    /// * `index` - The index of the vec to set the value at
+    /// * `value` - The value to be set
+    ///
+    /// # Reverts
+    /// 
+    /// Reverts if index is larger than the length of the vec
+    #[storage(read, write)]
+    pub fn set(self, index: u64, value: V) {
```

Comment:
@mohammadfawaz could the compiler somehow desguar into this from `storage.storage_vec[1] = 42;`?

Reply:
Not yet unfortunately. This requires the `Index` trait: https://github.com/FuelLabs/sway/issues/429

Also related: https://github.com/FuelLabs/sway/issues/1849